libparsec/crates/client/tests/unit/invite/shamir.rs

```diff
@@ -0,0 +1,248 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::{collections::HashMap, sync::Arc};
+
+use super::super::super::client::tests::utils::client_factory;
+use libparsec_protocol::invited_cmds::v4::invite_info::ShamirRecoveryRecipient;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use crate::{
+    claimer_retrieve_info, ClientConfig, MountpointMountStrategy, ProxyConfig,
+    ShamirRecoveryClaimMaybeRecoverDeviceCtx, UserOrDeviceClaimInitialCtx,
+    WorkspaceStorageCacheSize,
+};
+
+#[parsec_test(testbed = "shamir", with_server)]
+async fn shamir(tmp_path: TmpPath, env: &TestbedEnv) {
```

Comment:
This function is long, I would split it using sub-function

Reply:
I'm not convinced this simplifies the reading :)